xwiki-commons-core/xwiki-commons-displayer/src/main/java/org/xwiki/displayer/HTMLDisplayer.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.displayer;
+
+import java.util.Map;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Display an html element based on the type of {@link T}.
+ *
+ * @version $Id$
+ * @since 10.10RC1
+ */
+@Role
+@Unstable
+public interface HTMLDisplayer<T>
+{
+    /**
+     * @param parameters parameters used while generating the html. Could be the attributes of an input for instance.
+     * @return the html element related to the class
+     */
+    String display(Map<String, String> parameters);
```

Comment:
I think you misunderstood my previous comment. It's fine to have a display method without the custom parameters but as I said it should tale a value (as in `T value`) IMO since that's what the displayer is mainly about.

Reply:
@tmortagne I'm not sure to understand. You want something like this?

```suggestion
    <E> String display(E value);
```

Reply:
Ok so you want me to do:

```suggestion
    String display(T value, Map<String, String> parameters);
```

Reply:
For this method yes.

Reply:
@tmortagne I've made the change